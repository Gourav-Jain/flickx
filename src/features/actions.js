/**
 * Action Types
 */
export const SEARCH_MOVIES_REQS = "SEARCH_MOVIES_REQS";
export const SEARCH_MOVIES_RESP = "SEARCH_MOVIES_RESP";
export const SEARCH_MOVIES_FAIL = "SEARCH_MOVIES_FAIL";

export const ADD_TO_FAVOURITE = "ADD_TO_FAVOURITE";
export const ADD_TO_WATCH_LATER = "ADD_TO_WATCH_LATER";
export const REMOVE_FROM_FAVOURITE = "REMOVE_FROM_FAVOURITE";
export const REMOVE_FROM_WATCH_LATER = "REMOVE_FROM_WATCH_LATER";

/**
 * Action Creator
 */
export const setSearchMoviesRequest = payload => ({ type: SEARCH_MOVIES_REQS, payload });
export const setSearchMoviesResponse = payload => ({ type: SEARCH_MOVIES_RESP, payload });
export const setSearchMoviesFailer = payload => ({ type: SEARCH_MOVIES_FAIL, payload });

export const setAddToFavourites = payload => ({ type: ADD_TO_FAVOURITE, payload });
export const setAddToWatchLater = payload => ({ type: ADD_TO_WATCH_LATER, payload });
export const setRemoveFromFavourite = payload => ({ type: REMOVE_FROM_FAVOURITE, payload });
export const setRemoveFromWatchLater = payload => ({ type: REMOVE_FROM_WATCH_LATER, payload });