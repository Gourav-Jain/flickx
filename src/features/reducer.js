import {
    SEARCH_MOVIES_REQS,
    SEARCH_MOVIES_RESP,
    SEARCH_MOVIES_FAIL,
    ADD_TO_FAVOURITE,
    ADD_TO_WATCH_LATER,
    REMOVE_FROM_FAVOURITE,
    REMOVE_FROM_WATCH_LATER
} from "./actions";
import initialState from '../store/initialState';

export const reducer = (state = initialState, action) => {
    switch (action.type) {
        case SEARCH_MOVIES_REQS:
            return {
                ...state,
                ...action.payload
            }
        case SEARCH_MOVIES_RESP:
            const { favourites, watchLater } = state || [];
            let { searchResult } = action.payload;
            if (favourites) {
                favourites.forEach(el => {
                    const idx = searchResult.result.findIndex(i => i.id === el.id);
                    if (idx !== -1) {
                        searchResult.result[searchResult.result.findIndex(i => i.id === el.id)].fav = true;
                    }
                });
            }

            if (watchLater) {
                watchLater.forEach(el => {
                    const idx = searchResult.result.findIndex(i => i.id === el.id);
                    if (idx !== -1) {
                        searchResult.result[searchResult.result.findIndex(i => i.id === el.id)].wl = true;
                    }
                });
            }
            const newState = { ...state, ...action.payload }

            return newState;
        case SEARCH_MOVIES_FAIL:
            return { ...state, ...action.payload };
        case ADD_TO_FAVOURITE:
            return action.payload;
        case ADD_TO_WATCH_LATER:
            return action.payload;
        case REMOVE_FROM_FAVOURITE:
            return action.payload;
        case REMOVE_FROM_WATCH_LATER:
            return action.payload;

        default:
            return state;
    }
};