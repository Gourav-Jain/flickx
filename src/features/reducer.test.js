import { reducer as searchReducer } from "./reducer";
import * as actions from "./actions";

describe("Create app reducer", () => {
    test.each`
    name                              | reducerFn              | state           | mockData
    ${"sucessfull setSearchMoviesRequest"} | ${"setSearchMoviesRequest"} | ${null} | ${''}
    ${"sucessfull setSearchMoviesResponse"} | ${"setSearchMoviesResponse"} | ${null} | ${''}
    ${"sucessfull setSearchMoviesFailer"} | ${"setSearchMoviesFailer"} | ${null} | ${''}
    ${"sucessfull setAddToFavourites"} | ${"setAddToFavourites"} | ${null} | ${''}
    ${"sucessfull setAddToWatchLater"} | ${"setAddToWatchLater"} | ${null} | ${''}
    ${"sucessfull setRemoveFromFavourite"} | ${"setRemoveFromFavourite"} | ${null} | ${''}
    ${"sucessfull setRemoveFromWatchLater"} | ${"setRemoveFromWatchLater"} | ${null} | ${''}
  `("should return $name", ({ reducerFn, state, mockData }) => {
        const action = reducerFn ? actions[reducerFn](mockData) : {};
        expect(searchReducer(state, action)).toMatchSnapshot();
    });
});
