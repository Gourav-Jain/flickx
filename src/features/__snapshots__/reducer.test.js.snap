// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create app reducer should return sucessfull setAddToFavourites 1`] = `""`;

exports[`Create app reducer should return sucessfull setAddToWatchLater 1`] = `""`;

exports[`Create app reducer should return sucessfull setRemoveFromFavourite 1`] = `""`;

exports[`Create app reducer should return sucessfull setRemoveFromWatchLater 1`] = `""`;

exports[`Create app reducer should return sucessfull setSearchMoviesFailer 1`] = `Object {}`;

exports[`Create app reducer should return sucessfull setSearchMoviesRequest 1`] = `Object {}`;

exports[`Create app reducer should return sucessfull setSearchMoviesResponse 1`] = `Object {}`;
